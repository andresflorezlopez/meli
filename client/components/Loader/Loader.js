import React from "react";
import PropTypes from "prop-types";
import {StyledLoader, StyledLoaderContainer}from "./Loader.styled.js";

const Loader = ({ color }) => {
  return (
    <StyledLoaderContainer>
      <StyledLoader color={color} />
    </StyledLoaderContainer>
  );
};

Loader.propTypes = {
  color: PropTypes.string
};

Loader.defaultProps = {
  color: "secondary"
};

export default Loader;
