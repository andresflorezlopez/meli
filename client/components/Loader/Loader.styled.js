import styled from "styled-components";

export const StyledLoader = styled.div`
  display: inline-block;
  width: 80px;
  margin: 0 auto;
  height: 80px;
  &:after {
    content: " ";
    display: block;
    width: 64px;
    height: 64px;
    margin: 8px;
    border-radius: 50%;
    border: 6px solid ${({ color, theme }) => theme.colors[color]};
    border-color: ${({ color, theme }) =>
      `${theme.colors[color]} transparent ${theme.colors[color]} transparent`};
    animation: dual-ring 1.2s linear infinite;
  }
  @keyframes dual-ring {
    0% {
      transform: rotate(0deg);
    }
    100% {
      transform: rotate(360deg);
    }
  }
`;

export const StyledLoaderContainer = styled.div`
  width: 100%;
  height: 80vh;
  display: flex;
  justify-content: center;
  align-items: center;
`;
