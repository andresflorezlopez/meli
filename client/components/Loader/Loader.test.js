import React from "react";
import Loader from "./Loader";
import { mount } from "enzyme";

import renderer from "react-test-renderer";
import LightTheme from "@config/themes/light";
import { ThemeProvider } from "styled-components";

describe("Loader", () => {
  test("should render", () => {
    const component = renderer.create(
      <ThemeProvider theme={LightTheme}>
        <Loader color="secondary" />{" "}
      </ThemeProvider>
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
