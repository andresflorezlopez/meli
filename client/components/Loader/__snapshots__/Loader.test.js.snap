// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loader should render 1`] = `
Array [
  <div
    className="sc-1g2dfmx-1 iJYVVG"
  >
    <div
      className="sc-1g2dfmx-0 dFNVXP"
      color="secondary"
    />
  </div>,
  " ",
]
`;
